### PR TITLE
decompile SND_* files

### PR DESCRIPTION
**Pull request by ipatix**
All SND_*.s files decompiled. Code quality is not the cleanest, if there is anything that I should fix, please let me know.

### CODE REVIEW

File: arm9/lib/include/SND_interface.h [hunk 1]
```diff
@@ -0,0 +1,62 @@
+#ifndef GUARD_SND_INTERFACE_H
+#define GUARD_SND_INTERFACE_H
+
+#include "global.h"
+#include "SND_bank.h"
+#include "SND_work.h"
+
+void SND_StartSeq(int player, const void *seqBasePtr, u32 seqOffset, struct SNDBankData *bankData);
```
Comment (red031000) [inline arm9/lib/include/SND_interface.h:8]: s32 instead of int

File: arm9/lib/include/SND_main.h [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+#define GUARD_SND_MAIN_H
+
+#include "global.h"
+
+void SNDi_UnlockMutex(void);
+void SNDi_LockMutex(void);
+void SND_Init(void);
+
+struct SNDBinaryFileHeader {
+    char signature[4];  // 0x0
```
Comment (red031000) [inline arm9/lib/include/SND_main.h:11]: s8 instead of char

File: arm9/lib/src/SND_command.c [hunk 3]
```diff
@@ -0,0 +1,337 @@
...
+        return;
+
+    do {
+        OS_SpinWait(100);
+    } while (!PXI_IsCallbackReady(7, 1));
+}
+
+ARM_FUNC static void RequestCommandProc(void) {
+    while (PXI_SendWordByFifo(7, 0, 0) < 0)
+        ;
```
Comment (red031000) [inline arm9/lib/src/SND_command.c:308]: replace with { } to prevent warnings

File: arm9/lib/src/SND_interface.c [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+#define PushCmd1(id, a)          PushCommand_impl(id, (u32)(a), 0, 0, 0)
+#define PushCmd2(id, a, b)       PushCommand_impl(id, (u32)(a), (u32)(b), 0, 0)
+#define PushCmd3(id, a, b, c)    PushCommand_impl(id, (u32)(a), (u32)(b), (u32)(c), 0)
+#define PushCmd4(id, a, b, c, d) PushCommand_impl(id, (u32)(a), (u32)(b), (u32)(c), (u32)(d))
+
+// TODO fill in "random" constants with macros
+
+// void SND_StartSeq(int player, const void *seqBasePtr, u32 seqOffset, struct SNDBankData *bankData) { }
+
+void SND_StopSeq(int player) {
```
Comment (red031000) [inline arm9/lib/src/SND_interface.c:14]: ARM_FUNC

Comment (red031000) [reply arm9/lib/src/SND_interface.c:14]: also s32

File: arm9/lib/src/SND_main.c [hunk 5]
```diff
@@ -0,0 +1,28 @@
+#include "SND_main.h"
+
+#include "global.h"
+#include "OS_mutex.h"
+
+static struct OSMutex sSndMutex;
+static int sSndInitialized;
```
Comment (red031000) [inline arm9/lib/src/SND_main.c:7]: s32

File: arm9/lib/src/SND_util.c [hunk 6]
```diff
@@ -0,0 +1,236 @@
+#include "SND_util.h"
+
+// not used in pokediamond
+//static const u16 sPitchTable[0x600] = {
```
Comment (red031000) [inline arm9/lib/src/SND_util.c:4]: imo remove if not used

File: arm9/lib/src/SND_util.c [hunk 7]
```diff
@@ -0,0 +1,236 @@
...
+//     -99,  -96,  -94,  -91,  -89,  -87,  -85,  -82,
+//     -80,  -78,  -76,  -74,  -72,  -70,  -68,  -66,
+//     -64,  -62,  -60,  -58,  -56,  -54,  -52,  -50,
+//     -49,  -47,  -45,  -43,  -42,  -40,  -38,  -36,
+//     -35,  -33,  -31,  -30,  -28,  -27,  -25,  -23,
+//     -22,  -20,  -19,  -17,  -16,  -14,  -13,  -11,
+//     -10,   -8,   -7,   -6,  -4,   -3,   -1,     0,
+//};
+
+ARM_FUNC u16 SND_CalcChannelVolume(int x) {
```
Comment (red031000) [inline arm9/lib/src/SND_util.c:212]: s32

File: arm9/lib/src/SND_work.c [hunk 8]
```diff
@@ -0,0 +1,115 @@
+#include "SND_work.h"
+#include "SND_alarm.h"
+#include "SND_main.h"
+
+struct SNDSharedWork *SNDi_SharedWork;
+
+void DC_InvalidateRange(void *mem, u32 size);
+void DC_FlushRange(void *mem, u32 size);
+
+u32 SND_GetPlayerStatus(void) {
```
Comment (red031000) [inline arm9/lib/src/SND_work.c:10]: ARM_FUNC

File: arm9/lib/src/SND_work.c [hunk 9]
```diff
@@ -0,0 +1,115 @@
...
+//    DC_InvalidateRange(&SNDi_SharedWork->captureStatus, 2);
+//    return SNDi_SharedWork->captureStatus;
+//}
+
+u32 SND_GetPlayerTickCounter(u32 playerId) {
+    DC_InvalidateRange(&SNDi_SharedWork->players[playerId].tickCounter, 4);
+    return SNDi_SharedWork->players[playerId].tickCounter;
+}
+
+//s16 SND_GetPlayerLocalVariable(u32 playerId, u32 var) {
```
Comment (red031000) [inline arm9/lib/src/SND_work.c:30]: remove if not used

File: arm9/lib/src/SND_bank.c [hunk 10]
```diff
@@ -0,0 +1,175 @@
...
+            break;
+        }
+
+        instPos->program++;
+        instPos->index = 0;
+    }
+    return FALSE;
+}
+
+u32 SND_GetWaveDataCount(const struct SNDWaveArc *waveArc) {
```
Comment (red031000) [inline arm9/lib/src/SND_bank.c:152]: ARM_FUNC

**Comment by ipatix** [comment]
Updated, the commit.

**Comment by ipatix** [comment]
Updated once again.

**Pull request merged**